File: Contadordecafe/app/src/main/java/com/esampaio/apps/contadordecafe/view/activity/MainActivity.java
package com.esampaio.apps.contadordecafe.view.activity;

import android.content.Intent;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.MotionEvent;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.esampaio.apps.contadordecafe.view.adapter.ListCafeAdapter;
import com.esampaio.apps.contadordecafe.R;
import com.esampaio.apps.contadordecafe.persistence.RepositorioCafe;
import com.esampaio.apps.contadordecafe.view.estatistica.activity.ChartActivity;

public class MainActivity extends AppCompatActivity {

    private ImageView btBeberCafe;
    private TextView txtQtdCafeBebido;
    private TextView titulo;
    private TextView subtitulo;
    private RepositorioCafe repositorioCafe;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        repositorioCafe = new RepositorioCafe(this);
        btBeberCafe = (ImageView) findViewById(R.id.bt_beber_cafe);
        txtQtdCafeBebido = (TextView) findViewById(R.id.txt_qtd_cafe_hoje);
        subtitulo = (TextView) findViewById(R.id.subtitulo);
        btBeberCafe.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View v, MotionEvent event) {
                switch (event.getAction()) {
                    case MotionEvent.ACTION_DOWN:
                        btBeberCafe.setRotation(-45);
                        break;
                    case MotionEvent.ACTION_UP:
                        btBeberCafe.setRotation(0);
                        beberCafe();
                }
                return true;
            }
        });
        loadCafesBebidos();
        ActionBar actionBar = getSupportActionBar();
        if(actionBar!=null){
            actionBar.setTitle(getString(R.string.toolbar_name));
        }
    }

    private void loadCafesBebidos() {
        Integer qtdCafesBebidosHj = repositorioCafe.getQtdCafeHj();
        if(qtdCafesBebidosHj <= 1){
            subtitulo.setText("Café hoje");
        }else{
            subtitulo.setText("Cafés hoje");
        }
        txtQtdCafeBebido.setText(String.valueOf(qtdCafesBebidosHj));
    }
    private void beberCafe(){
        repositorioCafe.beberCafeHj();
        loadCafesBebidos();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater menuInflater = getMenuInflater();
        menuInflater.inflate(R.menu.main_menu,menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if(item.getItemId() == R.id.menu_graficos){
            Intent intent = new Intent(MainActivity.this,ChartActivity.class);
            startActivity(intent);
        }
        return super.onOptionsItemSelected(item);
    }
}

File: Contadordecafe/app/src/main/java/com/esampaio/apps/contadordecafe/view/activity/MainActivity.kt
package com.esampaio.apps.contadordecafe.view.activity

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import android.view.MotionEvent
import com.esampaio.apps.contadordecafe.R
import com.esampaio.apps.contadordecafe.persistence.RepositorioCafe
import com.esampaio.apps.contadordecafe.view.estatistica.activity.ChartActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private var repositorioCafe: RepositorioCafe? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        repositorioCafe = RepositorioCafe(this)

        bt_beber_cafe.setOnTouchListener { _, event ->
            when (event.action) {
                MotionEvent.ACTION_DOWN -> bt_beber_cafe!!.rotation = -45f
                MotionEvent.ACTION_UP -> {
                    bt_beber_cafe!!.rotation = 0f
                    beberCafe()
                }
            }
            true
        }
        loadCafesBebidos()
        val actionBar = supportActionBar
        if (actionBar != null) {
            actionBar.title = getString(R.string.toolbar_name)
        }
    }

    private fun loadCafesBebidos() {
        val qtdCafesBebidosHj = repositorioCafe?.qtdCafeHj
        if (qtdCafesBebidosHj!! <= 1) {
            subtitulo.text = "Café hoje"
        } else {
            subtitulo.text = "Cafés hoje"
        }
        txt_qtd_cafe_hoje!!.text = qtdCafesBebidosHj.toString()
    }

    private fun beberCafe() {
        repositorioCafe?.beberCafeHj()
        loadCafesBebidos()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val menuInflater = menuInflater
        menuInflater.inflate(R.menu.main_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.menu_graficos) {
            val intent = Intent(this@MainActivity, ChartActivity::class.java)
            startActivity(intent)
        }
        return super.onOptionsItemSelected(item)
    }
}
